MuscleModel/EMG/GetEMGInfo.m

```diff
@@ -7,122 +7,136 @@
 
 % Author: Maarten Afschrift
 
-boolEMG = 0;
-% check if the input is correct
-if isfield(Misc,'EMGconstr') && Misc.EMGconstr == 1
-    boolEMG = 1;
-    nF = length(Misc.EMGfile);
-    if nF ~=length(DatStore)
-        disp('Warning: number of EMG files is not equal to the number of IK or ID files.');
+if Misc.boolEMG
+    for trial = Misc.trials_sel
+        if isempty(Misc.EMGfile{trial}) || isempty(Misc.IKfile{trial}) || isempty(Misc.IDfile{trial})
+            warning(['EMG or IK or ID file for trial ' Misc.trialName{trial} 'has not been defined. Please update Misc.EMGfile, Misc.IKfile or Misc.IDfile']);
+        end
     end
 end
 
-
-if boolEMG    
+if Misc.boolEMG    
     % file information
-    nF = length(Misc.EMGfile);
+    bool_error  = 0;
+    IndError=zeros(length(Misc.EMGSelection),length(Misc.trials_sel));
     % Load the data and check for errors
-    for iF = 1:nF        
+    EMGFile = struct;
+    for t = 1:length(Misc.trials_sel)
+        iF = Misc.trials_sel(t);
         % get information for the EMG constraints
-        EMGFile(iF)      = ReadMotFile(Misc.EMGfile{iF});        
-    end    
-    % check if we have to update the headers based on user input
-    bool_updateheader   = 0;
-    if isfield(Misc,'EMGheaders') && ~isempty(Misc.EMGheaders)        
-        bool_updateheader=1;
-    end
-    % verify if the selected muscles are in the model
-    iF       = 1;    
-    bool_error  = 0;
-    IndError=zeros(length(Misc.EMGSelection),1);
-    for i=1:length(Misc.EMGSelection)
-        if ~any(strcmp(Misc.EMGSelection{i},DatStore(iF).MuscleNames))
-            disp(['Could not find ' Misc.EMGSelection{i} ' in the model, Update the Misc.EMGSelection structure']);
-            bool_error=1;
-            IndError(i)=1;
+        clear emgFile
+        emgFile = read_motionFile_v40(Misc.EMGfile{iF}); 
```

Comment:
why did you change ReadMotFile to read_motionFile_v40 ? Is this one faster (and if so, convince me we need an additional function)

Reply:
No big reason, honestly. The main reason to change it is that the version 4 of OpenSim has a new header variable called inDeg. This new function takes that into account. The inDeg variable is not important in this implementation. Both ReadMotFile and read_motionFile_v40 should work. The structures they output have the same data, but different names for column headers, just need to adjust that.